core/src/main/scala/org/apache/spark/ExecutorAllocationManager.scala

```diff
@@ -392,10 +397,36 @@ private[spark] class ExecutorAllocationManager(
   }
 
   /**
-   * Request the cluster manager to remove the given executor.
+   * Request the cluster manager to remove the given executors.
    * Return whether the request is received.
    */
-  private def removeExecutor(executorId: String): Boolean = synchronized {
+  private def removeExecutors(executorIds: Seq[String]): Boolean = synchronized {
+
+    val executorIdsToBeRemoved = executorIds.filter(canBeKilled)
+
+    // Send a request to the backend to kill this executor
+    val removeRequestAcknowledged = testing || client.killExecutors(executorIdsToBeRemoved)
```

Comment:
The problem is not the backend. The problem is `CoarseGrainedSchedulerBackend.killExecutors` which can decide to kill just a subset of the executors asked to be killed.

I'm wary of @dhruve's assessment of the problem because that code was added *exactly* because the code was killing busy executors that had for some reason become "idle".  See SPARK-9552.